server/src/main/java/org/elasticsearch/index/engine/LuceneChangesSnapshot.java

```diff
@@ -258,10 +258,19 @@ private TopDocs searchOperations(ScoreDoc after) throws IOException {
                 assert assertDocSoftDeleted(leaf.reader(), segmentDocID) : "Delete op but soft_deletes field is not set [" + op + "]";
             } else {
                 final BytesReference source = fields.source();
+                if (source == null) {
+                    if (requiredFullRange) {
```

Comment:
why is this tied to `requiredFullRange` so deep?  

PS - I thought we were heading towards always requiring full ranges? this make this problem moot.